plonky2x/core/src/frontend/vars/array.rs

```diff
@@ -180,7 +180,44 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
     /// Given an `array` of variables, and a dynamic `index` start_idx, returns
     /// `array[start_idx..start_idx+sub_array_size]` as an `array`.
     ///
-    /// `seed` is used to generate randomness for the proof.
+    /// `seed` is used to generate randomness for the proof. This function generates the
+    /// seed from the array and subarray, so the seed is not needed as an input.
+    pub fn get_fixed_subarray<const ARRAY_SIZE: usize, const SUB_ARRAY_SIZE: usize>(
+        &mut self,
+        array_bytes: &ArrayVariable<ByteVariable, ARRAY_SIZE>,
+        subarray_bytes: &ArrayVariable<ByteVariable, SUB_ARRAY_SIZE>,
+        start_idx: Variable,
+    ) -> ArrayVariable<Variable, SUB_ARRAY_SIZE> {
+        let array_variables = ArrayVariable::<Variable, ARRAY_SIZE>::from(
+            array_bytes
+                .as_slice()
+                .iter()
+                .map(|x| x.to_variable(self))
+                .collect_vec(),
+        );
+
+        let mut seed = Vec::new();
+        seed.extend_from_slice(array_bytes.as_slice());
```

Comment:
In this case, shouldn't the seed also use the `array_variables`? and potentially also the `subarray_bytes` converted to variables? otherwise there will be 8x the number of things being poseidon-ed

Reply:
In `get_fixed_subarray_unsafe`, it packs 7 `ByteVariable`'s into a single `Variable` (56 bits), as it knows what the size of the targets are. `array_variables` == `array_bytes`, just as `Variable`'s instead of `ByteVariable`'s.

If we were to pass in `Variable`'s as the seed to `get_fixed_subarray_unsafe`, there wouldn't be a simple way to pack the `seed`. 

The logic in `get_fixed_subarray_unsafe` that compresses the chunk of `ByteVariable`'s into a single `Variable` before poseidon hashing is here: https://github.com/succinctlabs/succinctx/blob/254cffcdd48d4e831eb70332032d4e324aff1d3e/plonky2x/core/src/frontend/vars/array.rs#L255-L265

Reply:
Oh I see, I didn't realize the packing was actually already going on in the unsafe function! that makes sense to me